### PR TITLE
MNT Update @methods on class docblocks

### PR DESCRIPTION
**Pull request by emteknetnz**
Issue https://github.com/silverstripe/.github/issues/163

Used short names instead of fqcn for relation types if class was imported

### CODE REVIEW

File: src/Security/Member.php [hunk 1]
```diff
@@ -57,6 +55,9 @@
  * @property int $FailedLoginCount
  * @property string $DateFormat
  * @property string $TimeFormat
+ * @method SilverStripe\ORM\ManyManyList<Group> Groups()
```
Comment (GuySartorelli) [inline src/Security/Member.php]:
```suggestion
 * @method ManyManyList<Group> Groups()
```
And in a few other classes as well - the class is imported but we're still using the FQCN

File: src/ORM/Hierarchy/Hierarchy.php [hunk 2]
```diff
@@ -24,7 +24,6 @@
  *
  * @property int $ParentID
  * @property DataObject|Hierarchy $owner
- * @method DataObject Parent()
```
Comment (GuySartorelli) [inline src/ORM/Hierarchy/Hierarchy.php]: See https://github.com/silverstripe/silverstripe-framework/pull/11092#discussion_r1424737200

**Review by GuySartorelli — CHANGES_REQUESTED** [review]

File: src/ORM/Hierarchy/Hierarchy.php (hunk 2 above)
Comment (emteknetnz) [reply src/ORM/Hierarchy/Hierarchy.php]: Good spot. I've done a search around looks like this is the only instance of get_extra_config() affecting relation config so manually fixing here is correct

File: src/Security/Member.php (hunk 1 above)
Comment (emteknetnz) [reply src/Security/Member.php]: Whoops, forgot to handle belongs_many_many in the code writer

File: src/ORM/Hierarchy/Hierarchy.php (hunk 2 above)
Comment (GuySartorelli) [reply src/ORM/Hierarchy/Hierarchy.php]: Sweet. I've added a note about it in the follow-up issue so CI doesn't flag it later on.

**Review by GuySartorelli — COMMENTED** [review]
Don't forget to check for `belongs_to`

**Comment by emteknetnz** [comment]
Have updated code writer

**Review by GuySartorelli — APPROVED** [review]
LGTM

**Pull request merged**